independent-projects/qute/core/src/main/java/io/quarkus/qute/LoopSectionHelper.java

```diff
@@ -312,6 +312,60 @@ public CompletionStage<Object> getAsync(String key) {
             };
         }
 
+        /**
+         * This method exists for the purpose of saving the cost of creating a substring of the key and
+         * computing its hashCode.
+         */
+        private CompletionStage<Object> getCompletedStage(String key, int keyStartIndex, int count) {
```

Comment:
Hm, I see what you mean but I wonder if it would make more sense to precompute the metadata keys, store it in an array (in a field of `LoopSectionHelper`) and replace this stuff with a simple if-then-else. I'll try to prepare something so that we can compare the results...

Reply:
If we can have something which comparing is as easy as comparing among enums - would be even better

Reply:
The thing is that the `alias` (which is part of the prefix) can be different for a `{#for}` definition...